File: CMS.Models/Models/KhachDatHang.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace CMS.Models
{
    [Table("KhachDatHang")]
    public class KhachDatHang
    {
        [Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Required]
        public int KhachDatHangID { get; set; }

        public int? NguoiDungID { get; set; }
        public int? UserID { get; set; }

        [Column(TypeName = "Date")]
        public DateTime? NgayDat { get; set; }

        public int? GioDat { get; set; }

        public string GhiChu { get; set; }

        public int? ThanhTien { get; set; }
        public int? TinhTrangXuLy { get; set; }
        public int? HinhThucThanhToan { get; set; }
        public int? TinhTrangThanhToan { get; set; }

        [ForeignKey("NguoiDungID")]
        public virtual NguoiDung NguoiDung { get; set; }
        [ForeignKey("UserID")]
        public virtual Users Users { get; set; }

        [InverseProperty("KhachDatHang")]
        public virtual ICollection<ChiTietDoGiat> ChiTietDoGiat { get; set; }

    }
}

File: CMS.Admin/src/models/KhachDatHang.ts
import { ChiTietDoGiat } from "@/models/ChiTietDoGiat";
import { NguoiDung } from "@/models/NguoiDung";
import { Users } from "./Users";

export interface KhachDatHang {
    KhachDatHangID: number;
    NguoiDungID: number;
    NgayDat: Date;
    GioDat: number;
    GhiChu: string;
    ThanhTien: number;
    TinhTrangXuLy: number;
    HinhThucThanhToan: number;
    TinhTrangThanhToan: number;
    UserID: number;
    NguoiDung?: NguoiDung;
    Users?: Users;
    ChiTietDoGiat?: ChiTietDoGiat[];
}
